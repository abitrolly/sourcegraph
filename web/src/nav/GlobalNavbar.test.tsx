import React from 'react'
import { setLinkComponent } from '../../../shared/src/components/Link'
import * as GQL from '../../../shared/src/graphql/schema'
import { ThemePreference } from '../theme'
import { GlobalNavbar } from './GlobalNavbar'
import { createLocation, createMemoryHistory } from 'history'
import { NOOP_SETTINGS_CASCADE } from '../../../shared/src/util/searchTestHelpers'
import { mount } from 'enzyme'

const PROPS: GlobalNavbar['props'] = {
    authenticatedUser: null,
    extensionsController: {} as any,
    location: createLocation('/'),
    history: createMemoryHistory(),
    hideGlobalSearchInput: false,
    keyboardShortcuts: [],
    isSourcegraphDotCom: false,
    navbarSearchQueryState: { query: 'q', cursorPosition: 0 },
    onNavbarQueryChange: () => undefined,
    lowProfile: false,
    onThemePreferenceChange: () => undefined,
    isLightTheme: true,
    themePreference: ThemePreference.Light,
    patternType: GQL.SearchPatternType.literal,
    setPatternType: () => undefined,
    caseSensitive: false,
    setCaseSensitivity: () => undefined,
    platformContext: {} as any,
    settingsCascade: NOOP_SETTINGS_CASCADE,
    showCampaigns: false,
    telemetryService: {} as any,
    hideNavLinks: true, // used because reactstrap Popover is incompatible with react-test-renderer and enzyme
    filtersInQuery: {} as any,
    splitSearchModes: false,
    interactiveSearchMode: false,
    toggleSearchMode: () => undefined,
    onFiltersInQueryChange: () => undefined,
    smartSearchField: false,
    isSearchRelatedPage: true,
    copyQueryButton: false,
    versionContext: undefined,
    setVersionContext: () => undefined,
    availableVersionContexts: [],
}

describe('GlobalNavbar', () => {
    setLinkComponent(props => <a {...props} />)
    afterAll(() => setLinkComponent(() => null)) // reset global env for other tests

    test('normal', () => expect(mount(<GlobalNavbar {...PROPS} />)).toMatchSnapshot())

    test('lowProfile', () => expect(mount(<GlobalNavbar {...PROPS} lowProfile={true} />)).toMatchSnapshot())

    test('hideGlobalSearchInput', () =>
        expect(mount(<GlobalNavbar {...PROPS} hideGlobalSearchInput={true} />)).toMatchSnapshot())
})
