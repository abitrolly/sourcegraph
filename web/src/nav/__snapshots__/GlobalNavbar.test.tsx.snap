// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalNavbar hideGlobalSearchInput 1`] = `
<GlobalNavbar
  authenticatedUser={null}
  availableVersionContexts={Array []}
  caseSensitive={false}
  copyQueryButton={false}
  extensionsController={Object {}}
  filtersInQuery={Object {}}
  hideGlobalSearchInput={true}
  hideNavLinks={true}
  history="[History]"
  interactiveSearchMode={false}
  isLightTheme={true}
  isSearchRelatedPage={true}
  isSourcegraphDotCom={false}
  keyboardShortcuts={Array []}
  location="[Location path=/]"
  lowProfile={false}
  navbarSearchQueryState={
    Object {
      "cursorPosition": 0,
      "query": "q",
    }
  }
  onFiltersInQueryChange={[Function]}
  onNavbarQueryChange={[Function]}
  onThemePreferenceChange={[Function]}
  patternType="literal"
  platformContext={Object {}}
  setCaseSensitivity={[Function]}
  setPatternType={[Function]}
  setVersionContext={[Function]}
  settingsCascade={
    Object {
      "final": null,
      "subjects": null,
    }
  }
  showCampaigns={false}
  smartSearchField={false}
  splitSearchModes={false}
  telemetryService={Object {}}
  themePreference="light"
  toggleSearchMode={[Function]}
>
  <div
    className="global-navbar global-navbar--bg border-bottom py-1"
  >
    <div
      className="global-navbar__logo-link global-navbar__logo-animated"
    >
      <img
        className="global-navbar__logo"
        src="/.assets/img/sourcegraph-mark.svg"
      />
    </div>
    <div
      className="nav-item flex-1"
    >
      <Component
        className="nav-link"
        to="/search"
      >
        <a
          className="nav-link"
          to="/search"
        >
          Search
        </a>
      </Component>
    </div>
  </div>
</GlobalNavbar>
`;

exports[`GlobalNavbar lowProfile 1`] = `
<GlobalNavbar
  authenticatedUser={null}
  availableVersionContexts={Array []}
  caseSensitive={false}
  copyQueryButton={false}
  extensionsController={Object {}}
  filtersInQuery={Object {}}
  hideGlobalSearchInput={false}
  hideNavLinks={true}
  history="[History]"
  interactiveSearchMode={false}
  isLightTheme={true}
  isSearchRelatedPage={true}
  isSourcegraphDotCom={false}
  keyboardShortcuts={Array []}
  location="[Location path=/]"
  lowProfile={true}
  navbarSearchQueryState={
    Object {
      "cursorPosition": 0,
      "query": "q",
    }
  }
  onFiltersInQueryChange={[Function]}
  onNavbarQueryChange={[Function]}
  onThemePreferenceChange={[Function]}
  patternType="literal"
  platformContext={Object {}}
  setCaseSensitivity={[Function]}
  setPatternType={[Function]}
  setVersionContext={[Function]}
  settingsCascade={
    Object {
      "final": null,
      "subjects": null,
    }
  }
  showCampaigns={false}
  smartSearchField={false}
  splitSearchModes={false}
  telemetryService={Object {}}
  themePreference="light"
  toggleSearchMode={[Function]}
>
  <div
    className="global-navbar  py-1"
  >
    <div
      className="flex-1"
    />
  </div>
</GlobalNavbar>
`;

exports[`GlobalNavbar normal 1`] = `
<GlobalNavbar
  authenticatedUser={null}
  availableVersionContexts={Array []}
  caseSensitive={false}
  copyQueryButton={false}
  extensionsController={Object {}}
  filtersInQuery={Object {}}
  hideGlobalSearchInput={false}
  hideNavLinks={true}
  history="[History]"
  interactiveSearchMode={false}
  isLightTheme={true}
  isSearchRelatedPage={true}
  isSourcegraphDotCom={false}
  keyboardShortcuts={Array []}
  location="[Location path=/]"
  lowProfile={false}
  navbarSearchQueryState={
    Object {
      "cursorPosition": 0,
      "query": "q",
    }
  }
  onFiltersInQueryChange={[Function]}
  onNavbarQueryChange={[Function]}
  onThemePreferenceChange={[Function]}
  patternType="literal"
  platformContext={Object {}}
  setCaseSensitivity={[Function]}
  setPatternType={[Function]}
  setVersionContext={[Function]}
  settingsCascade={
    Object {
      "final": null,
      "subjects": null,
    }
  }
  showCampaigns={false}
  smartSearchField={false}
  splitSearchModes={false}
  telemetryService={Object {}}
  themePreference="light"
  toggleSearchMode={[Function]}
>
  <div
    className="global-navbar global-navbar--bg border-bottom py-1"
  >
    <div
      className="global-navbar__logo-link global-navbar__logo-animated"
    >
      <img
        className="global-navbar__logo"
        src="/.assets/img/sourcegraph-mark.svg"
      />
    </div>
  </div>
</GlobalNavbar>
`;
